app/cron.server.ts

```diff
@@ -124,9 +124,35 @@ const scheduleUpdateEricaRequest = (cronExpression: string) => {
   });
 };
 
+export const deleteOutdatedSurveyResults = async () => {
+  try {
+    const now = new Date();
+    const oneDayAgo = new Date(now.setUTCDate(now.getUTCDate() - 1));
+    const queryResult = await db.survey.deleteMany({
+      where: {
+        createdAt: {
+          lte: oneDayAgo,
+        },
+      },
```

Comment:
what do you think if we could delete the data between two dates instead?

first survey creation is an absolute date, so we could do something like this?
`SELECT * FROM "Survey" WHERE "Survey"."createdAt" BETWEEN <Date_of_First_Survey> AND <OneDayAgo>`

So we do not need to delete old data manually. Do you see any issue with this logic?

Reply:
Maybe something like this?

https://github.com/prisma/prisma/discussions/11443#discussioncomment-2062783

```js
{
    where: {
      createdAt: {
        lte: oneDayAgo,
        gte: firstDateOfSurvey,
      },
    }
}
```

Reply:
>what do you mean? is it running every minute? hour?
This cronjob would not run only once a day. However, I think this would still be fine. wdyt?
start-cron.ts where the Cron function is called. why don't we call the cleanup in there?

This are exactly the right questions. I totally forgot to add it to start-cron.ts. Therefore, it is not specified how often they are run. I added it to be run once a day. Thanks for catching that 👍

Reply:
> what do you think if we could delete the data between two dates instead?
> [...]
>So we do not need to delete old data manually. Do you see any issue with this logic? 

That would work, too. However, we would definitely only delete the entries for the last day and would keep potential older data. This would also mean that if the cronjob would not run once, it would not delete anything.